addons/pos_loyalty/static/src/overrides/components/payment_screen/payment_screen.js

```diff
@@ -80,7 +80,10 @@ patch(PaymentScreen.prototype, {
                 points: pe.points - order._getPointsCorrection(ProgramModel.get(pe.program_id)),
             });
             const program = ProgramModel.get(pe.program_id);
-            if (program.is_nominative && partner) {
+            if (
+                (program.is_nominative || program.program_type == "next_order_coupons") &&
+                partner
+            ) {
```

Comment:
What this change will do?
It does not look like it'll set partner on `next_order_coupons` :thinking: